include/Options/Options.php

```diff
@@ -533,4 +538,51 @@ public function offsetUnset( $offset ): void {
 	private function is_locked( int $blog_id ): bool {
 		return isset( $this->modified[ $blog_id ] ) && false === $this->modified[ $blog_id ];
 	}
+
+	/**
+	 * Returns the list of modified sites.
+	 * On multisite, sites are cached.
+	 * /!\ At this point, some sites may not appear as locked while they have been deleted: this can happen if they are
+	 * not deleted with `wp_delete_site()`. Those sites are locked here.
+	 *
+	 * @since 3.7
+	 *
+	 * @return bool[]
+	 * @phpstan-return array<int, true>
+	 */
+	private function get_modified(): array {
```

Comment:
I'm not sure this method is really needed.
Instead we could use `get_site()` as we said.
It looks like `WP_Site::get_instance()` hits the cache and we can bet that is a blog is in our modified options it has been fetched already (so the cache is primed).
Also, even if the cache is empty and the query is made, since it's done on `shutdown` it's not a big deal IMHO.

Reply:
But maybe you'd like an opinion from @Chouby?

Reply:
This is to lower the chances to call `get_sites()` because I think `wp_delete_site()` will be used in most cases.

Reply:
Simplified in [264175a](https://github.com/polylang/polylang/pull/1498/commits/264175a61160b67511fdbfb89b9cd7ccb0c0a386).